src/test/java/uk/gov/register/serialization/CommandParserTest.java

```diff
@@ -26,66 +28,75 @@
+        assertTrue(registerCommands.next() instanceof AddItemCommand);
+        assertTrue(registerCommands.hasNext());
+        assertTrue(registerCommands.next() instanceof AppendEntryCommand);
     }
 
-    @Test
-    public void shouldParseAppendEntryCommand() {
-        RegisterCommand registerCommand = commandParser.newCommand("append-entry\t2016-11-02T14:45:54Z\tsha-256:22c3e102937320cb613660e88116bb937c1873ed54321966ee089e1d71fb4b7f");
-        assertTrue(registerCommand instanceof AppendEntryCommand);
-    }
-
-
     @Test(expected = SerializedRegisterParseException.class)
     public void shouldThrowExWhenNoHash() throws Exception {
         String line = "append-entry\t2016-10-12T17:45:19.757132";
-        commandParser.newCommand(line);
+        commandParser.addCommand(line);
     }
 
     @Test(expected = SerializedRegisterParseException.class)
     public void shouldThrowExWhenNoContent() throws Exception {
         String line = "add-item";
-        commandParser.newCommand(line);
+        commandParser.addCommand(line);
     }
 
     @Test(expected = SerializedRegisterParseException.class)
     public void shouldThrowExWhenNoHashPrefix() throws Exception {
         String line = "append-entry\t2016-10-12T17:45:19.757132\tabc123";
-        commandParser.newCommand(line);
+        commandParser.addCommand(line);
     }
 
     @Test(expected = SerializedRegisterParseException.class)
     public void shouldThrowExWhenUnknownCommand() throws Exception {
         String line = "unknown-command-here\tabc123";
-        commandParser.newCommand(line);
+        commandParser.addCommand(line);
     }
 
     @Test(expected = DateTimeParseException.class)
     public void shouldFailIfTimestampNotIso() throws Exception {
         String line = "append-entry\t20161212\tsha-256:abc123";
-        commandParser.newCommand(line);
+        commandParser.addCommand(line);
     }
 
     @Test(expected = SerializedRegisterParseException.class)
     public void shouldFailIfInvalidJson() throws Exception {
         String line = "add-item\t{\"address\":\"9AQZJ3K\"";
-        commandParser.newCommand(line);
+        commandParser.addCommand(line);
+    }
+
+    @Test(expected = OrphanItemException.class)
+    public void shouldFailForOrphanItem() throws Exception {
+        commandParser.addCommand("add-item\t{\"register\":\"ft_openregister_test\",\"text\":\"SomeText\"}");
+        commandParser.addCommand("add-item\t{\"register\":\"ft_openregister_test\",\"text\":\"orphan\"}");
+        commandParser.addCommand("append-entry\t2016-11-02T14:45:54Z\tsha-256:3cee6dfc567f2157208edc4a0ef9c1b417302bad69ee06b3e96f80988b37f254");
+        commandParser.getCommands();
+    }
+
+    @Test(expected = OrphanItemException.class)
+    public void shouldFailForItemAfterEntry() throws Exception {
+        commandParser.addCommand("append-entry\t2016-11-02T14:45:54Z\tsha-256:3cee6dfc567f2157208edc4a0ef9c1b417302bad69ee06b3e96f80988b37f254");
```

Comment:
I didn't think 'dangling' is very clear, so I introduced the term 'orphan' item for an item which no entry refers to. There are actually 2 parts to the requirement viz the item should be referred to by an entry and that entry must come later in the RSF.